### DISCUSSION TITLE
Sender name display in group threads in addition to first initial icon?

### DISCUSSION THREAD

**Discussion by steveodat** [FossifyOrg/General-Discussion#500] [Q&A] [answered]
So, first off, great project, thanks so much for making this happen.

In Messages, in a group chat, the only indication of who has sent each message is the user image (or in my case, a generated first-letter-of-the-contact-name icon).  

When multiple people have the same first initial, there's no differentiation between them. It all looks like it's coming from the same person. (Screenshot included of such a thread, with two people with the same first letter of name: 
![Sameletter](https://github.com/user-attachments/assets/8118cd14-e45b-4a46-b720-8555a2957c49) )

Is there a way to display the sender name in small text for each message above or below the message bubble, just in threads with more than one receiver?  Or some other way to include the sender's name?  Not sure if I'm missing a setting somewhere, but I'm not finding it, and from a brief perusal of the code, I don't see where this is an option?

If not, I guess move this to Ideas/Feature Requests as a priority.

And again, thanks!

**Comment by naveensingh** [comment]
Currently, there is no way to change the automatically generated thumbnail. I'll think about how this can be improved before the next release.

**Reply by steveodat** [reply]
I was thinking more along the lines of: 
![Screenshot_20240903-140149_1](https://github.com/user-attachments/assets/2ef8446b-405b-4d35-88cd-5a883f515256)
with the name below.  (Obviously placement and size and color is a rough up.)

Or better yet!  The name above, that way you only have to display it if it doesn't match the contact of the previous message.  So messages sent in a row from the same person would appear to be grouped under that name.
e.g.: 
![Screenshot_20240903-140149_2](https://github.com/user-attachments/assets/d9cf5d84-1eaf-40f4-8edb-41e08b7c223f)

Android app dev isn't my area, but from that gander at the code, would you add it as an additional item in the list prior to the message, like you do when the last message sent time is more than 300 seconds old?  Or is there a way to display the contact name _with_ the message item in the list? (Again, extremely not my coding skills area.)

**Comment by Aga-C** [comment, marked as answer]
There's already a request for this: https://github.com/FossifyOrg/Messages/issues/81.

**Reply by steveodat** [reply]
Oh dang!  I looked for just such an open issue before posting, but somehow completely did not see it!  Well, a +1 to that.  And thanks!
